packages/material-ui/src/Select/SelectInput.js

```diff
@@ -369,6 +369,20 @@ const SelectInput = React.forwardRef(function SelectInput(props, ref) {
       foundMatch = true;
     }
 
+    const hasOptionSelected = () => {
+      if (arr.find((item) => areEqualValues(value, item.props.value))) {
+        return selected;
+      }
+      return true;
+    };
+
+    if (child.props.value === undefined) {
```

Comment:
This returns a different object when the option is a ListSubheader. The role option is because the role listbox of the Menu needs that the children have the role option, but when the option is not a selectable option the aria-selected should be omitted.